File: c/model/me_components.c
#include <stdlib.h>
#include <string.h>
#include <stdio.h>
#include "me_components.h"


char* verticalme_components_ToString(one_msg_waba_sdk_me_components_VERTICAL_e vertical) {
    char* verticalArray[] =  { "NULL", "Automotive", "Beauty, Spa and Salon", "Clothing and Apparel", "Education", "Entertainment", "Event Planning and Service", "Finance and Banking", "Food and Grocery", "Public Service", "Hotel and Lodging", "Medical and Health", "Non-profit", "Professional Services", "Shopping and Retail", "Travel and Transportation", "Restaurant", "Other", "" };
	return verticalArray[vertical];
}

one_msg_waba_sdk_me_components_VERTICAL_e verticalme_components_FromString(char* vertical){
    int stringToReturn = 0;
    char *verticalArray[] =  { "NULL", "Automotive", "Beauty, Spa and Salon", "Clothing and Apparel", "Education", "Entertainment", "Event Planning and Service", "Finance and Banking", "Food and Grocery", "Public Service", "Hotel and Lodging", "Medical and Health", "Non-profit", "Professional Services", "Shopping and Retail", "Travel and Transportation", "Restaurant", "Other", "" };
    size_t sizeofArray = sizeof(verticalArray) / sizeof(verticalArray[0]);
    while(stringToReturn < sizeofArray) {
        if(strcmp(vertical, verticalArray[stringToReturn]) == 0) {
            return stringToReturn;
        }
        stringToReturn++;
    }
    return 0;
}

me_components_t *me_components_create(
    char *about,
    char *address,
    char *description,
    char *email,
    char *phone,
    one_msg_waba_sdk_me_components_VERTICAL_e vertical,
    char *photo
    ) {
    me_components_t *me_components_local_var = malloc(sizeof(me_components_t));
    if (!me_components_local_var) {
        return NULL;
    }
    me_components_local_var->about = about;
    me_components_local_var->address = address;
    me_components_local_var->description = description;
    me_components_local_var->email = email;
    me_components_local_var->phone = phone;
    me_components_local_var->vertical = vertical;
    me_components_local_var->photo = photo;

    return me_components_local_var;
}


void me_components_free(me_components_t *me_components) {
    if(NULL == me_components){
        return ;
    }
    listEntry_t *listEntry;
    if (me_components->about) {
        free(me_components->about);
        me_components->about = NULL;
    }
    if (me_components->address) {
        free(me_components->address);
        me_components->address = NULL;
    }
    if (me_components->description) {
        free(me_components->description);
        me_components->description = NULL;
    }
    if (me_components->email) {
        free(me_components->email);
        me_components->email = NULL;
    }
    if (me_components->phone) {
        free(me_components->phone);
        me_components->phone = NULL;
    }
    if (me_components->photo) {
        free(me_components->photo);
        me_components->photo = NULL;
    }
    free(me_components);
}

cJSON *me_components_convertToJSON(me_components_t *me_components) {
    cJSON *item = cJSON_CreateObject();

    // me_components->about
    if(me_components->about) {
    if(cJSON_AddStringToObject(item, "about", me_components->about) == NULL) {
    goto fail; //String
    }
    }


    // me_components->address
    if(me_components->address) {
    if(cJSON_AddStringToObject(item, "address", me_components->address) == NULL) {
    goto fail; //String
    }
    }


    // me_components->description
    if(me_components->description) {
    if(cJSON_AddStringToObject(item, "description", me_components->description) == NULL) {
    goto fail; //String
    }
    }


    // me_components->email
    if(me_components->email) {
    if(cJSON_AddStringToObject(item, "email", me_components->email) == NULL) {
    goto fail; //String
    }
    }


    // me_components->phone
    if(me_components->phone) {
    if(cJSON_AddStringToObject(item, "phone", me_components->phone) == NULL) {
    goto fail; //String
    }
    }


    // me_components->vertical
    if(me_components->vertical != one_msg_waba_sdk_me_components_VERTICAL_NULL) {
    if(cJSON_AddStringToObject(item, "vertical", verticalme_components_ToString(me_components->vertical)) == NULL)
    {
    goto fail; //Enum
    }
    }


    // me_components->photo
    if(me_components->photo) {
    if(cJSON_AddStringToObject(item, "photo", me_components->photo) == NULL) {
    goto fail; //String
    }
    }

    return item;
fail:
    if (item) {
        cJSON_Delete(item);
    }
    return NULL;
}

me_components_t *me_components_parseFromJSON(cJSON *me_componentsJSON){

    me_components_t *me_components_local_var = NULL;

    // me_components->about
    cJSON *about = cJSON_GetObjectItemCaseSensitive(me_componentsJSON, "about");
    if (about) { 
    if(!cJSON_IsString(about) && !cJSON_IsNull(about))
    {
    goto end; //String
    }
    }

    // me_components->address
    cJSON *address = cJSON_GetObjectItemCaseSensitive(me_componentsJSON, "address");
    if (address) { 
    if(!cJSON_IsString(address) && !cJSON_IsNull(address))
    {
    goto end; //String
    }
    }

    // me_components->description
    cJSON *description = cJSON_GetObjectItemCaseSensitive(me_componentsJSON, "description");
    if (description) { 
    if(!cJSON_IsString(description) && !cJSON_IsNull(description))
    {
    goto end; //String
    }
    }

    // me_components->email
    cJSON *email = cJSON_GetObjectItemCaseSensitive(me_componentsJSON, "email");
    if (email) { 
    if(!cJSON_IsString(email) && !cJSON_IsNull(email))
    {
    goto end; //String
    }
    }

    // me_components->phone
    cJSON *phone = cJSON_GetObjectItemCaseSensitive(me_componentsJSON, "phone");
    if (phone) { 
    if(!cJSON_IsString(phone) && !cJSON_IsNull(phone))
    {
    goto end; //String
    }
    }

    // me_components->vertical
    cJSON *vertical = cJSON_GetObjectItemCaseSensitive(me_componentsJSON, "vertical");
    one_msg_waba_sdk_me_components_VERTICAL_e verticalVariable;
    if (vertical) { 
    if(!cJSON_IsString(vertical))
    {
    goto end; //Enum
    }
    verticalVariable = verticalme_components_FromString(vertical->valuestring);
    }

    // me_components->photo
    cJSON *photo = cJSON_GetObjectItemCaseSensitive(me_componentsJSON, "photo");
    if (photo) { 
    if(!cJSON_IsString(photo) && !cJSON_IsNull(photo))
    {
    goto end; //String
    }
    }


    me_components_local_var = me_components_create (
        about && !cJSON_IsNull(about) ? strdup(about->valuestring) : NULL,
        address && !cJSON_IsNull(address) ? strdup(address->valuestring) : NULL,
        description && !cJSON_IsNull(description) ? strdup(description->valuestring) : NULL,
        email && !cJSON_IsNull(email) ? strdup(email->valuestring) : NULL,
        phone && !cJSON_IsNull(phone) ? strdup(phone->valuestring) : NULL,
        vertical ? verticalVariable : -1,
        photo && !cJSON_IsNull(photo) ? strdup(photo->valuestring) : NULL
        );

    return me_components_local_var;
end:
    return NULL;

}

File: typescript-fetch/src/models/MeComponents.ts
/* tslint:disable */
/* eslint-disable */
/**
 * one_msg_waba_sdk
 * [1MSG.IO](https://1msg.io/) is the perfect WhatsApp management tool for your business. With us you get full access to the official Whatsapp API/webhooks.  Every API request must contain an Authorize HTTP header with a token.  This is your channel token, which can be found in your channel project on your profile page. Please do not give the token to anyone or post it publicly.  The authorization token must be added to each request in the GET parameter \'token\' and always passed to query string (?token={your_token}). Parameters in GET queries pass query string. Parameters in POST requests — through the JSON-encoded request body.   All \'send\' methods (except /sendTemplate) will only work when the dialog session with the user is open. Some of our solutions simplify and avoid such limitations, but we urge you to pay more attention to this detail
 *
 * The version of the OpenAPI document: 2.39.0
 * Contact: hello@1msg.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface MeComponents
 */
export interface MeComponents {
    /**
     * Profile`s About section.
     * @type {string}
     * @memberof MeComponents
     */
    about?: string;
    /**
     * Address of the business
     * @type {string}
     * @memberof MeComponents
     */
    address?: string;
    /**
     * Description of the business. Max 256 characters
     * @type {string}
     * @memberof MeComponents
     */
    description?: string;
    /**
     * Business email
     * @type {string}
     * @memberof MeComponents
     */
    email?: string;
    /**
     * Linked phone number
     * @type {string}
     * @memberof MeComponents
     */
    phone?: string;
    /**
     * Industry of the business
     * @type {string}
     * @memberof MeComponents
     */
    vertical?: MeComponentsVerticalEnum;
    /**
     * HTTP link
     * *https://...*
     * 
     * Or base64-encoded file with mime data, for example
     * *data:image/jpeg;base64,/9j/4AAQSkZJRgABAQ...* 
     * 
     * File in form-data input field
     * @type {string}
     * @memberof MeComponents
     */
    photo?: string;
}


/**
 * @export
 */
export const MeComponentsVerticalEnum = {
    Automotive: 'Automotive',
    BeautySpaAndSalon: 'Beauty, Spa and Salon',
    ClothingAndApparel: 'Clothing and Apparel',
    Education: 'Education',
    Entertainment: 'Entertainment',
    EventPlanningAndService: 'Event Planning and Service',
    FinanceAndBanking: 'Finance and Banking',
    FoodAndGrocery: 'Food and Grocery',
    PublicService: 'Public Service',
    HotelAndLodging: 'Hotel and Lodging',
    MedicalAndHealth: 'Medical and Health',
    NonProfit: 'Non-profit',
    ProfessionalServices: 'Professional Services',
    ShoppingAndRetail: 'Shopping and Retail',
    TravelAndTransportation: 'Travel and Transportation',
    Restaurant: 'Restaurant',
    Other: 'Other',
    Empty: ''
} as const;
export type MeComponentsVerticalEnum = typeof MeComponentsVerticalEnum[keyof typeof MeComponentsVerticalEnum];


/**
 * Check if a given object implements the MeComponents interface.
 */
export function instanceOfMeComponents(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function MeComponentsFromJSON(json: any): MeComponents {
    return MeComponentsFromJSONTyped(json, false);
}

export function MeComponentsFromJSONTyped(json: any, ignoreDiscriminator: boolean): MeComponents {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'about': !exists(json, 'about') ? undefined : json['about'],
        'address': !exists(json, 'address') ? undefined : json['address'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'email': !exists(json, 'email') ? undefined : json['email'],
        'phone': !exists(json, 'phone') ? undefined : json['phone'],
        'vertical': !exists(json, 'vertical') ? undefined : json['vertical'],
        'photo': !exists(json, 'photo') ? undefined : json['photo'],
    };
}

export function MeComponentsToJSON(value?: MeComponents | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'about': value.about,
        'address': value.address,
        'description': value.description,
        'email': value.email,
        'phone': value.phone,
        'vertical': value.vertical,
        'photo': value.photo,
    };
}

